mailgun/resource_mailgun_credential.go

```diff
@@ -0,0 +1,217 @@
+package mailgun
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/mailgun/mailgun-go/v3"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type mailgunCredential struct {
+	Email    string
+	Region   string
+	Password string
+}
+
+func resourceMailgunCredential() *schema.Resource {
+	log.Printf("[DEBUG] resourceMailgunCredential()")
+
+	return &schema.Resource{
+		CreateContext: resourceMailgunCredentialCreate,
+		Read:          resourceMailgunCredentialRead,
+		Update:        resourceMailgunCredentialUpdate,
+		Delete:        resourceMailgunCredentialDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceMailgunCredentialImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"login": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"password": {
+				Type:      schema.TypeString,
+				ForceNew:  false,
+				Required:  true,
+				Sensitive: true,
+			},
+
+			"domain": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"region": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
+				Default:  "us",
+			},
+		},
+	}
+}
+
+func resourceMailgunCredentialImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.SplitN(d.Id(), ":", 2)
```

Comment:
The logic in this method is a bit complicated, could you look at the logic on domin importing?